### PR TITLE
www/caddy: Add authentik auth provider

### PR DESCRIPTION
**Pull request by Tim-Sc**
Add support for the authentik auth provider.
See: https://docs.goauthentik.io/docs/providers/proxy/server_caddy

### CODE REVIEW

**Comment by Monviech** [comment]
Nice, gonna check it out soon and give feedback. I guess it was a good choice to make these providers pluggable, I didn't include Authentik since I had no way to test this. Thanks. :)

**Comment by Monviech** [comment]
Are the headers in Authentik always the same requirement and can be hardcoded? Because if they are configuration dependant, it might be a good choice to implement a way to change headers again. I have already changed the header menu once to allow copy headers with a lot of additional logic.

I can help with that if it is necessary.

https://github.com/Monviech/opnsense-plugins/commit/6825f51b3e65f010ac201d6558a258ae6f7899e3#diff-c905c5709a75510121b03ead89789a33e45e9514b7e7facd591d8bcdf503263b

**Comment by Monviech** [comment]
@Tim-Sc Good job, here is some feedback from me.

Could we maybe rewrite the `includeAuthProvider` file like this? That way it will be clearer in the future when more providers are added.

<details>

```
{#
# This file gets imported to configure forward auth in handlers.
# - Section: Reverse Proxy Configurations
#}
{% if generalSettings.AuthProvider %}
    {# Check if the domain is IPv6 and wrap in square brackets if necessary #}
    {% set is_ipv6 = (':' in generalSettings.AuthToDomain and generalSettings.AuthToDomain.count(':') >= 2) %}
    {% set auth_url = (generalSettings.AuthToTls|default("0") == "1" and 'https://' or 'http://') + (is_ipv6 and '[' or '') + generalSettings.AuthToDomain|default("") + (is_ipv6 and ']' or '') + (generalSettings.AuthToPort and ':' + generalSettings.AuthToPort or '') %}
{% endif %}
{% if generalSettings.AuthProvider == 'authelia' %}
    forward_auth {{ auth_url }} {
    {% if generalSettings.AuthToUri %}
        uri {{ generalSettings.AuthToUri|default("") }}
    {% endif %}
    copy_headers Remote-User Remote-Groups Remote-Name Remote-Email
    }
{% elif generalSettings.AuthProvider == 'authentik' %}
    reverse_proxy /outpost.goauthentik.io/* {{ auth_url }} {
    {% if generalSettings.AuthToTls|default("0") == "1" %}
        header_up Host {http.reverse_proxy.upstream.hostport}
    {% endif %}
    }
    forward_auth {{ auth_url }} {
    {% if generalSettings.AuthToUri %}
        uri {{ generalSettings.AuthToUri|default("") }}
    {% endif %}
    copy_headers X-Authentik-Username X-Authentik-Groups X-Authentik-Email X-Authentik-Name X-Authentik-Uid X-Authentik-Jwt X-Authentik-Meta-Jwks X-Authentik-Meta-Outpost X-Authentik-Meta-Provider X-Authentik-Meta-App X-Authentik-Meta-Version
    }
{% endif %}
```

</details>

The help text in the GUI should be adjusted:

https://github.com/opnsense/plugins/blob/9a200512a19d121592fcef9f1cdc018d87b050db/www/caddy/src/opnsense/mvc/app/controllers/OPNsense/Caddy/forms/authprovider.xml#L6

The pkg description should receive a new version number and what has changed. Put yourself as (contributed by XXX) next to the new feature. Here's an example:

https://github.com/opnsense/plugins/blob/9a200512a19d121592fcef9f1cdc018d87b050db/www/caddy/pkg-descr#L143

Please bump the plugin version to 1.6.2

https://github.com/opnsense/plugins/blob/9a200512a19d121592fcef9f1cdc018d87b050db/www/caddy/Makefile#L2

**Review by Monviech — APPROVED** [review]
LGTM

**Comment by Monviech** [comment]
@Tim-Sc Thank you for your contribution.

I won't implement custom headers without any reason to do so. :)

File: www/caddy/pkg-descr [hunk 1]
```diff
@@ -25,6 +25,10 @@ DOC: https://docs.opnsense.org/manual/how-tos/caddy.html
 Plugin Changelog
 ================
 
+1.6.2
+
+* Add: Authentik as authentication provider. [contributed by @Tim-Sc]
```
Comment (fichtner) [inline www/caddy/pkg-descr:30]:
```suggestion
* Add Authentik as authentication provider (contributed by Tim-Sc)
```

just a little style and github handles are meaningless when showing this in the GUI

File: www/caddy/src/opnsense/service/templates/OPNsense/Caddy/includeAuthProvider [hunk 2]
```diff
@@ -2,12 +2,28 @@
...
+        header_up Host {http.reverse_proxy.upstream.hostport}
+    {% endif %}
+    }
+    forward_auth {{ auth_url }} {
+    {% if generalSettings.AuthToUri %}
+        uri {{ generalSettings.AuthToUri|default("") }}
+    {% endif %}
+    copy_headers X-Authentik-Username X-Authentik-Groups X-Authentik-Email X-Authentik-Name X-Authentik-Uid X-Authentik-Jwt X-Authentik-Meta-Jwks X-Authentik-Meta-Outpost X-Authentik-Meta-Provider X-Authentik-Meta-App X-Authentik-Meta-Version
+    }
+{% endif %}
```
Comment (fichtner) [inline www/caddy/src/opnsense/service/templates/OPNsense/Caddy/includeAuthProvider:29]:
```suggestion
{% endif %}

```

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**